### PR TITLE
add tests for permissions when creating reference

### PR DESCRIPTION
**Pull request by juhanikat**

### CODE REVIEW

**Review by iritmaximus — APPROVED** [review]
LGTM

File: backend/src/api-tests/reference/create.test.ts [hunk 1]
```diff
@@ -29,8 +29,22 @@ describe('Creating new reference works', () => {
 
     const { status: getReqStat } = await send<ReferenceDetailsType>(`reference/${createdId}`, 'GET')
     expect(getReqStat).toEqual(200)
+  })
+
+  it('Creation fails without permissions', async () => {
+    logout()
+    const { body: resultBody1, status: getReqStatus1 } = await send('reference/', 'PUT', {
```
Comment (iritmaximus) [inline backend/src/api-tests/reference/create.test.ts:36]: Small thing, but could these resultBody1 be named by the permissions it has? For example `resultBodyNoPerm` and `resultBodyEr`.

**Review by iritmaximus — CHANGES_REQUESTED** [review]
Sorry, I came up with a small fix.

**Comment by juhanikat** [comment]
created new pull request for this: #420

**Pull request closed without merging**